Fix Empty Delegations Request

deguzmankevin opened the pull request:
- ~Moves `delegations` variable to be accessible to other components to check if there are delegations pending, rather than having delegations be defined when submitting the form~
- Prevents API call to `DelegationsController` when there are no delegations so there's no `403` error
- Change the `postDelegations` code to follow standardized API calls
- ~Prevents `Submit Delegations` from being clicked if no delegations to be saved~

Fixes #239

kevintli commented on app/javascript/containers/ReviewModeContainer.jsx:
```diff
@@ -3,33 +3,22 @@
  */
 
-import React from 'react';
+import React from "react";
```
Prettier++ :D

kevintli commented on app/javascript/containers/ReviewModeContainer.jsx:
```diff
@@ -108,55 +71,97 @@ class ReviewModeContainer extends React.Component {
+        }
+      });
+    }
+    let empty = delegations.length == 0 ? true : false;
```
Oops, this is probably our fault for calling the method `submitDelegation`, but in the future we actually want the Submit Form button to take the user back to their home page and mark all of their delegated questions as completed (so they don't see them again). So even if the user clicks the button with no delegations to make, it should still succeed -- it just doesn't need to make the API call to submit new delegations. Let's keep the button enabled at all times, but add a check in `submitDelegations` to skip the request if there's nothing to do!

kevintli commented on app/javascript/containers/ReviewModeContainer.jsx:
```diff
@@ -108,55 +71,97 @@ class ReviewModeContainer extends React.Component {
 
-
   render() {
+    var parentQuestionsForDelegations = this.props.questions.filter(
```
Can you actually move this code for finding delegations and all their dependent questions to a separate function, and call it inside of `submitDelegations` instead of `render`? I'm worried that `render` will get called any time a prop updates, which would cause us to repeat all this fancy question finding logic unnecessarily instead of only running it once when the Submit button is clicked

deguzmankevin replied:
omg I knew there was something sketchy about this LOL I should've realized it!! good call

kevintli commented on app/javascript/containers/ReviewModeContainer.jsx:
```diff
@@ -50,54 +39,28 @@ function mapCategorytoQuestions(categoryMap, categoryId, building) {
+  async submitDelegations(delegations) {
+    this.setState({ status_string: "Saving delegations!" });
+    try {
+      let response = await post("/api/delegations", { delegations });
```
(See my below comment in `render()` first, but this is the line of code that doesn't need to run if there are no delegations to make)

deguzmankevin commented:
Looks like what you had in mind was what I did at first too! I changed the code back to my first commit and  moved the delegations code into a new function :)
